src/Http/WebUtilities/src/MultipartReaderStream.cs

```diff
@@ -147,7 +147,15 @@ private int UpdatePosition(int read)
             _observedLength = _position;
             if (LengthLimit.HasValue && _observedLength > LengthLimit.GetValueOrDefault())
```

Comment:
```suggestion
            if (LengthLimit.HasValue &&
                LengthLimit.GetValueOrDefault() is lengthLimit
                _observedLength > lengthLimit)
```

Then use `lengthLimit` elsewhere?